modules/xmpp/ChatRoom.js

```diff
@@ -430,6 +432,29 @@ export default class ChatRoom extends Listenable {
 
         // process nodes to extract data needed for MUC_JOINED and
         // MUC_MEMBER_JOINED events
+        const extractIdentityInformation = node => {
+            const identity = {};
+            const userInfo = node.children.find(c => c.tagName === 'user');
+
+            if (userInfo) {
+                identity.user = {};
+                for (const tag of [ 'id', 'name', 'avatar' ]) {
+                    const child
+                        = userInfo.children.find(c => c.tagName === tag);
+
+                    if (child) {
+                        identity.user[tag] = child.value;
+                    }
+                }
+            }
+            const groupInfo = node.children.find(c => c.tagName === 'group');
```

Comment:
If this is a node, would it be possible to use built in selectors to find the first child instead of iterating?

Reply:
this isn't an actual `node`, `node` is just the term they used when transforming the xml to json.